segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ReadOperation.java

```diff
@@ -120,6 +121,7 @@ private void logEnd() {
     }
 
     private CompletableFuture<Void> readData(MetadataTransaction txn) {
+        val chunkReadFutures = new ArrayList<CompletableFuture<Void>>();
```

Comment:
This collection is not thread safe yet it is passed around from thread to thread. Even if you're not modifying it concurrently, different threads may not pick up the changes from previous threads. Please use `Collections.synchronizedList` to wrap it.